core/src/main/kotlin/org/sourcegrade/jagr/core/compiler/java/JavaRuntimeResources.kt

```diff
@@ -19,34 +19,39 @@
 
 package org.sourcegrade.jagr.core.compiler.java
 
+import org.sourcegrade.jagr.api.testing.RuntimeResources
 import org.sourcegrade.jagr.launcher.io.ResourceContainer
 import org.sourcegrade.jagr.launcher.io.SerializationScope
 import org.sourcegrade.jagr.launcher.io.SerializerFactory
 import org.sourcegrade.jagr.launcher.io.keyOf
 import org.sourcegrade.jagr.launcher.io.readMap
 import org.sourcegrade.jagr.launcher.io.writeMap
+import java.util.*
 
-data class RuntimeResources(
+data class JavaRuntimeResources(
   val classes: Map<String, CompiledClass> = mapOf(),
   val resources: Map<String, ByteArray> = mapOf(),
-) {
-  companion object Factory : SerializerFactory<RuntimeResources> {
-    val base = keyOf<RuntimeResources>("base")
-    override fun read(scope: SerializationScope.Input) = RuntimeResources(scope.readMap(), scope.readMap())
+) : RuntimeResources {
+  override fun getClassNames(): Set<String> = Collections.unmodifiableSet(HashSet(classes.keys))
+  override fun getResourceNames(): Set<String> = Collections.unmodifiableSet(HashSet(resources.keys))
```

Comment:
Is it still necessary to wrap with `HashSet(...)`? `classes.keys` already returns a set.